react-client-app/src/components/Home.jsx

```diff
@@ -101,20 +101,34 @@ const Home = () => {
         // sort scan reports
         scanreports = scanreports.sort((b, a) => (a.id > b.id) ? 1 : ((b.id > a.id) ? -1 : 0))
         // create a list of unique datapartners and make a batch query to get their data
-        const dataPartnerObject = {}
+        const datasetObject = {};
         scanreports.map(scanreport => {
-            dataPartnerObject[scanreport.data_partner] = true
+            datasetObject[scanreport.parent_dataset] = true;
         })
-        const dataPartnerIds = chunkIds(Object.keys(dataPartnerObject))
-        const dataPartnerPromises = []
+        const datasetIds = Object.keys(datasetObject)
+        const datasetPromises = [];
+        for (let i = 0; i < datasetIds.length; i++) {
+            datasetPromises.push(useGet(`/datasets/${datasetIds[i]}`));
+        }
+        let datasets = await Promise.all(datasetPromises)
+        datasets = [].concat.apply([], datasets)
+        datasets.forEach((element) => {
+            scanreports = scanreports.map((scanreport) => scanreport.parent_dataset == element.id ? { ...scanreport, parent_dataset: element } : scanreport);
```

Comment:
what's this doing? i can see it's some sort of dict shuffle, but some code comments would be useful

Reply:
The Promise.all is waiting for all the requests to finish and returns the results, but it returns it as an array of arrays with the actual array we want being at index 0 so "datasets = [].concat.apply([], datasets)" just flattens the outer array.

The foreach loop then loops through the datasets that have just been retrieved and puts them into their respective scanreport objects

Reply:
cool thanks